app/app_xdg.go

```diff
@@ -122,5 +157,40 @@ func rootCacheDir() string {
 }
 
 func watchTheme() {
-	// no-op, not able to read linux theme in a standard way
+	go watchFreedekstopThemeChange()
+}
+
+func themeChanged() {
+	fyne.CurrentApp().Settings().(*settings).setupTheme()
+}
+
+// connect to dbus to detect color-schem theme changes in portal settings.
+func watchFreedekstopThemeChange() {
+	conn, err := dbus.SessionBus()
+	if err != nil {
+		log.Println("Unable to connect to session D-Bus", err)
+		return
+	}
+
+	if err := conn.AddMatchSignal(
+		dbus.WithMatchObjectPath("/org/freedesktop/portal/desktop"),
+		dbus.WithMatchInterface("org.freedesktop.portal.Settings"),
+		dbus.WithMatchMember("SettingChanged"),
+	); err != nil {
+		log.Println(err)
+		return
+	}
+	defer conn.Close()
+
+	dbusChan := make(chan *dbus.Signal, 10)
```

Comment:
Is there any specific reason that this has a length 10 buffer? Do we need as much?

Reply:
Changed, it seems to work actually with chan size to 0